src/dmd/dcast.d

```diff
@@ -1049,6 +1049,10 @@ MATCH implicitConvTo(Expression e, Type t)
         // Look for pointers to functions where the functions are overloaded.
         if (typeb.ty == Tdelegate && tb.ty == Tdelegate)
         {
+            if (!MODimplicitConv(tb.mod, typeb.mod) &&    // covariant check
+                global.params.useDIP1000 == FeatureState.enabled)  // because legacy code
```

Comment:
This is unrelated to scope pointers, and we're trying to get rid of these DIP1000 branches for the transition to deprecations in v2.101.

Reply:
Considering https://github.com/dlang/dmd/pull/14070 is in the same boat, maybe they can both go under a `-preview=fix-immutable-conv` switch.

Reply:
The main benefit of separate switches is easier migration. If `-preview=dip1000` only enables `scope` checks, then users who want to update their code for DIP 1000 compliance can do so without *also* having to make a bunch of unrelated updates at the same time.

A separate switch is also easier to document (since it will get its own help text) and more discoverable (since it will show up in `-preview=help`).